src/models/streaming/BufferedDuplex.ts

```diff
@@ -0,0 +1,53 @@
+import { Duplex, PassThrough } from 'stream';
+
+/**
+ * Represents a two way buffered stream. Extends Duplex which implements both Readable and Writable
+ */
+export class BufferedDuplex extends Duplex {
+    input: PassThrough;
+    output: PassThrough;
+    areOutputHandlersSetup: boolean;
+
+    /**
+     * @param input Buffer for the Writable side of the stream.
+     * @param output Buffer for the Readable side of the stream.
+     * @param options Options to be passed through to the superclass.
+     */
+    constructor (input: PassThrough, output: PassThrough, options?: any) {
+        super(options);
+        this.input = input;
+        this.output = output;
+        this.areOutputHandlersSetup = false;
+    }
+
+    _write(chunk: any, encoding: any, callback: any): boolean {
```

Comment:
should we put types on some of these parameters?
* chunk?
* encoding 'string'
* callback as `any` is perfectly fine

Reply:
Chunk has no type since it can be either a buffer or not